event/subscription.go

```diff
@@ -143,7 +143,7 @@ func (s *resubscribeSub) loop() {
 }
 
 func (s *resubscribeSub) subscribe() Subscription {
-	subscribed := make(chan error)
+	subscribed := make(chan error, 1)
```

Comment:
I think it would be better to fix this case by waiting on the `subscribed` channel when the `subscribe` loop
exits. This way, `Unsubscribe` will block until the goroutine running `s.fn` has exited.

Reply:
Actually, I have now applied the fix myself.